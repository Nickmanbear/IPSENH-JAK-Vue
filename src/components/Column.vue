<template>
  <div id="column">
    <h2>Title</h2>
    <div id="cards">
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
    </div>
    <p>+ Add card</p>
  </div>
</template>

<script>
// @ is an alias to /src
import Card from '@/components/CardPreview.vue';

export default {
  name: 'Home',
  components: {
    Card,
  },
};
</script>

<style lang="scss">
  #column {
    background-color: #eee;
    border: 1px solid #eee;
    border-radius: 3px;
    padding: 5px 10px;
    margin: 5px;
    display: inline-block;
    width: 240px;
    max-height: 80vh;
  }

  h2 {
    margin: 5px 0;
  }

  #cards {
    height: 70vh;
    overflow: scroll;
    -ms-overflow-style: none;
  }

  #cards::-webkit-scrollbar {
    display: none;
  }
</style>
