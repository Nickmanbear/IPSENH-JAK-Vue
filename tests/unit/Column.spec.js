import { shallowMount } from '@vue/test-utils';
import Column from '@/components/Column.vue';

jest.mock('axios', () => ({
  create: () => null,
  get: () => Promise.resolve({ data: [] }),
}));

describe('Column.spec.js', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(Column, {
      propsData: {
        column: {
          id: 6,
          boardId: 1,
          name: 'test column',
        },
      },
    });
  });

  it('equals name to "test column"', () => {
    expect(cmp.find('h2').text()).toEqual('test column');
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
