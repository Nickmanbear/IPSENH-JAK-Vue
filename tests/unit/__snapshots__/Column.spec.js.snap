// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column.spec.js has the expected html structure 1`] = `
<div
  id="column"
>
  <h2>
    test column
  </h2>
   
  <!---->
   
  <button>
    X
  </button>
   
  <div
    id="cards"
  >
    <draggable-stub
      clone="function(t){return t}"
      element="div"
      group="cards"
      value=""
    />
  </div>
   
  <button>
    + Add card
  </button>
   
  <!---->
</div>
`;
